src/backend/context.rs

```diff
@@ -488,6 +490,35 @@ impl Context {
 		Ok(())
 	}
 
+	/// Zoom a window.
+	fn zoom_window(&mut self, window_id: WindowId, zoom_delta: f32) -> Result<(), InvalidWindowId> {
+		let window = self
+			.windows
+			.iter_mut()
+			.find(|w| w.id() == window_id)
+			.ok_or(InvalidWindowId { window_id })?;
+
+		window.zoom += zoom_delta;
+		window.uniforms.mark_dirty(true);
+		window.window.request_redraw();
+		Ok(())
+	}
+
+	/// Pan a window.
+	fn pan_window(&mut self, window_id: WindowId, pan_x: f32, pan_y: f32) -> Result<(), InvalidWindowId> {
```

Comment:
Same as with `zoom_window`, I think just setting the pan value is better. I think we a solution to allow efficient access to read/write multiple window properties without doing a lookup each time.

Reply:
How do you propose to do it? I have to know the current uniforms (zoom and pan) in order to zoom or pan further right? That involves a lookup or some sort of getter.

Reply:
Yeah it's a bit problematic currently, because there are no cheap window handles. Every use of a window ID goes through the main context which has to look it up in the window list again.

Implementing such a handle isn't easy though, since we can't have a mutable borrow of the main context and a window at the same time (because the windows are owned by the main context). Without the main context there's a lot of things you can't do, but maybe we'll just have to accept that.